### PR TITLE
Ignores comparing log header format version to current format version

### PR DESCRIPTION
**Pull request by tinwelint**
since that version is no longer used, what matter now is the log entry
version, i.e. the version per entry, and that version only.

Also bumps the 2.2.4 log entry version to -4 to not collide and confuse
people with that of a 2.3 milestone.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/LogHeaderReader.java [hunk 1]
```diff
@@ -30,7 +30,6 @@
 import org.neo4j.kernel.impl.transaction.log.ReadableLogChannel;
 
 import static org.neo4j.kernel.impl.transaction.log.entry.LogHeader.LOG_HEADER_SIZE;
-import static org.neo4j.kernel.impl.transaction.log.entry.LogVersions.CURRENT_FORMAT_VERSION;
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/LogHeaderReader.java:33]: Can we delete the class `LogVersions`?

File: community/kernel/src/test/java/org/neo4j/kernel/impl/transaction/log/ReadMultipleLogEntryVersionsTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package org.neo4j.kernel.impl.transaction.log;
```
Comment (davidegrohmann) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/transaction/log/ReadMultipleLogEntryVersionsTest.java]: license header missing

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/LogHeaderReader.java (hunk 1 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/LogHeaderReader.java:33]: Let's wait, shall we? We're still writing that header into log files... we'll have to figure out when we remove that from the header as well.

**Comment by davidegrohmann** [comment]
@tinwelint we have a failing test which looks relevant

**Comment by digitalstain** [comment]
@davidegrohmann @tinwelint Just noting that the test passes locally for me. This will require more than a superficial investigation.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/LogHeaderReader.java (hunk 1 above)
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/LogHeaderReader.java:33]: If that constant is used only for writing log header, you can just move it there. I don't see the point of keeping that class around.

**Comment by tinwelint** [comment]
It was a problem with a badly created db, my bad. Also importantly @digitalstain when merging this into 2.3 you must change the 2_3 version in LogEntryVersion to `-5`

**Pull request merged**